src/backend/routes/mod.rs

```diff
@@ -1,54 +1,104 @@
-use std::fs;
+use std::sync::Arc;
 
+use gotham::helpers::http::response::create_response;
+use gotham::hyper::{Body, Response, StatusCode};
+use gotham::middleware::state::StateMiddleware;
+use gotham::pipeline::single::single_pipeline;
+use gotham::pipeline::single_middleware;
 use gotham::router::builder::*;
 use gotham::router::Router;
-use gotham::hyper::{Body, Response, StatusCode};
-use gotham::state::State;
-use gotham::helpers::http::response::create_response;
+use gotham::state::{FromState, State};
 use mime;
+use serde_json::{self, json};
 
-mod analysis;
+use crate::analyzer::Stats;
 
+/// Used to share analysis results across execution threads
+#[derive(Debug, Clone, StateData)]
+struct SharedHeapdump {
```

Comment:
This is not sharing the heapdump, but stats, should we rename it?

Reply:
Sure, good point!